include/real/real_data.hpp

```diff
@@ -288,6 +288,70 @@ namespace boost {
+
+                    bool exponent_is_even = false;
+
+                    if (remainder.empty()) {
+                        exponent_is_even = true;
+                    }
+
+                    if (ro.get_lhs_itr().get_interval().positive()) {
+                        this->_approximation_interval.upper_bound = 
+                                tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().upper_bound, exponent);
+                        this->_approximation_interval.lower_bound =
+                                tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().lower_bound, exponent);
+                    } else if (ro.get_lhs_itr().get_interval().negative()) {
+                        if (exponent_is_even) {
+                            this->_approximation_interval.upper_bound =
+                                    tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().lower_bound, exponent);
+                            this->_approximation_interval.lower_bound =
+                                    tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().upper_bound, exponent);
+                        } else {
+                            this->_approximation_interval.upper_bound =
+                                    tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().upper_bound, exponent);
+                            this->_approximation_interval.lower_bound =
+                                    tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().lower_bound, exponent);
+                        }
+                    } else {
+                        if (exponent_is_even) {
+                            if (ro.get_lhs_itr().get_interval().upper_bound.abs() > ro.get_lhs_itr().get_interval().lower_bound.abs()) {
+                                this->_approximation_interval.upper_bound =
+                                        tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().upper_bound, exponent);
+                                this->_approximation_interval.lower_bound = zero;
+                            } else {
+                                this->_approximation_interval.upper_bound =
+                                        tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().lower_bound, exponent);
+                                this->_approximation_interval.lower_bound = zero;
+                            }
+                        } else {
+                            this->_approximation_interval.upper_bound =
+                                    tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().upper_bound, exponent);
+                            this->_approximation_interval.lower_bound =
+                                    tmp.binary_exponentiation(ro.get_lhs_itr().get_interval().lower_bound, exponent);
+                        }
+                    }
+
+                    break;
+                }
```

Comment:
The "power" function you created to return an explicit number, for integer power of an explicit number, is never used.
I think you should use std::visit to specify this operation for different types of numbers. As, we will have to make this method available to integer and rational number types also, so at the end, we will have to specify this for different, instead of using intervals everywhere. So, having one more specification for "explicit_number", to return an "explicit_number" instead of operation acting on intervals everytime, is not bad.

Reply:
Yeah earlier I was using a visiter to implement power differently for each type of real number but then after discussing with Laouen I changed the implementation to work with intervals which is more general. We forgot that integer_number and rational_number don't use intervals. So we need to implement power separately for integers and rationals. I'm not sure if I can do this without using a visiter. May be mentors can suggest something better @Laouen @sdavtaker

Reply:
All real numbers should expose a precision iterator that iterate its boundaries, and intenger and rational should too (I think we agree on leting the integer as a backend structure, then no need to implement power for integers).

Reply:
Hi, As I mentioned in a previos comment, I think that the solution is that rational should implement the real numbers interface, which consist basically of having a precision iterator which is a forward iterator that iterate the rational intervals, where in each iteration, that interval is smaller, converging to the represented rational number.